src/test/fixtures/mocks/date.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// This function makes it possible to mock the Date object to return a specific
+// date when it's not specified.
+
+export function mockDate(dateOrTimestamp: string | number) {
+  const originalDate = global.Date;
+
+  jest.spyOn(global, 'Date').mockImplementation(function(...args) {
+    if (!args.length) {
+      return new Date(dateOrTimestamp);
+    }
+    return new originalDate(...args);
+  });
+
+  // We need to define the hasInstance well-known symbol, because global.Date is
+  // a function (a jest mock) otherwise. By defining this property, we support
+  // the `instanceof` operator that some of our code uses (especially
+  // fake-indexeddb)
+  Object.defineProperty(global.Date, Symbol.hasInstance, {
+    value: function(val) {
+      return val instanceof originalDate;
+    },
+  });
+
+  global.Date.now = jest
+    .fn()
+    .mockImplementation(() => new Date(dateOrTimestamp).valueOf());
+  global.Date.UTC = originalDate.UTC;
+  global.Date.parse = originalDate.parse;
```

Comment:
I think it would be worth looping over the prototype, and implementing functions that throw errors when called, notifiying that the mock wasn't implementing. This would remove any mysterious "undefined is not a function" errors.

`Object.getOwnPropertyNames(Date.prototype)`

Reply:
As discussed on matrix, this isn't needed.